File: dashboard_managers/dashy/dashy_section.rb
class DashySection

  attr_reader :name
  attr_reader :icon
  attr_reader :items

  def initialize(name, icon, items)
    @name = name
    @icon = icon
    @items = items
  end

  def as_json(options={})
    {
      name: @name,
      icon: @icon,
      items: @items
    }
  end

  def to_json(*options)
    as_json(*options).to_json(*options)
  end

  # def to_s
  #   "DashySection:\n" +
  #   "  name: #{@name}\n" +
  #   "  icon: #{@icon}\n" +
  #   "  items: #{@items}\n"
  # end

end

File: dotnet/Dsm.Managers/DashboardManagers/Dashy/DashySection.cs
namespace Dsm.Managers.Dashy;
public class DashySection
{
    public DashySection(string name, string? icon, List<DashyItem> items)
    {
        Name = name;
        Icon = icon;
        Items = items;
    }

    [Obsolete("For serialization")]
    public DashySection()
    {
    }

    public string? Name { get; set; }
    public string? Icon { get; set; }
    public List<DashyItem> Items { get; set; } = new List<DashyItem>();
}